infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationLogService.kt

```diff
@@ -132,9 +133,10 @@ constructor(
     }
 
     @Transactional(readOnly = true)
-    fun fetchLatestPublicationDetails(branchType: LayoutBranchType, count: Int): List<PublicationDetails> {
-        return publicationDao.fetchLatestPublications(branchType, count).map { getPublicationDetails(it.id) }
-    }
+    fun fetchLatestPublicationDetails(branchType: LayoutBranchType, count: Int): Page<PublicationDetails> =
+        publicationDao.list(branchType).let {
+            Page(it.size, it.take(count), 0).map { item -> getPublicationDetails(item.id) }
+        }
```

Comment:
Pihviä. Listaus palauttaa kaikki, mutta sivulle otetaan mukaan vain `count`:in kertoma määrä. Kokonaismäärä palautetaan nyt kuitenkin oikein.

Reply:
Niin et kuitenkaan lähtenyt toteuttamaan sitä että se lista olisi oikeasti sivutettu? Senhän saisi tuossa samalla, mutta toki enemmän fronttijumppaa.

Tuo on muuten varsin höhelö että se listaa nuo kaikki ja sitten vielä hakee kullekin erikseen tuon getPublicationDetails-kutsun, joka puolestaan sisältää kasan kanta-roundtrippejä per rivi. Eli jos tässä tulee mitään perf-ongelmaa niin tuo alkuperäinen listaus ei ole sen ongelma vaan tämän n+1 logiikka (joka siis oli kyllä jo aiemmin). Tuo funkkari itse asiassa hakee vielä sen saman publication-olionkin uudelleen tuolla id:llä jonka on ottanut juuri ulos samasta oliosta :D Toki tuo on vain hassu, isompi impakti varmasti sillä että se tekee kasan round trippejä joikaisella getPublicationDetails-kutsulla.

```
        val publication = publicationDao.getPublication(id)
        val ratkoStatus = ratkoPushDao.getRatkoStatus(id).sortedByDescending { it.endTime }.firstOrNull()

        val publishedReferenceLines = publicationDao.fetchPublishedReferenceLines(id)
        val publishedKmPosts = publicationDao.fetchPublishedKmPosts(id)
        val (publishedDirectTrackNumbers, publishedIndirectTrackNumbers) = publicationDao.fetchPublishedTrackNumbers(id)
        val (publishedDirectTracks, publishedIndirectTracks) = publicationDao.fetchPublishedLocationTracks(id)
        val (publishedDirectSwitches, publishedIndirectSwitches) = publicationDao.fetchPublishedSwitches(id)
        val split = splitService.getSplitIdByPublicationId(id)?.let(splitService::get)

```